torch/csrc/inductor/aoti_eager/kernel_meta_info.cpp

```diff
@@ -0,0 +1,88 @@
+#if !defined(C10_MOBILE) && !defined(ANDROID)
+#include <torch/csrc/inductor/aoti_eager/kernel_meta_info.h>
+
+namespace torch::inductor {
+
+TensorMetaInfo::TensorMetaInfo(const at::Tensor& src_tensor)
+    : is_symbolic_(false),
+      device_(src_tensor.device()),
+      sizes_(src_tensor.sym_sizes().vec()),
+      strides_(src_tensor.sym_strides().vec()) {
+  for (const auto& size : sizes_) {
+    if (size.is_symbolic()) {
+      is_symbolic_ = true;
+      break;
+    }
+  }
+
+  if (!is_symbolic_) {
+    for (const auto& stride : strides_) {
+      if (stride.is_symbolic()) {
+        is_symbolic_ = true;
+        break;
+      }
+    }
+  }
+
+  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(
+      !is_symbolic_,
+      "Eager through torch.compile does not support symbolic shape now.");
+}
+
+TensorMetaInfo::TensorMetaInfo(
+    bool is_symbolic,
+    c10::ScalarType dtype,
+    c10::Device device,
+    std::vector<c10::SymInt> sizes,
+    std::vector<c10::SymInt> strides)
+    : is_symbolic_(is_symbolic),
+      dtype_(dtype),
+      device_(device),
+      sizes_(sizes),
+      strides_(strides) {
+  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(
+      !is_symbolic_, "Not support symbolic shape now");
+}
+
+bool TensorMetaInfo::operator==(const TensorMetaInfo& other) const {
+  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(
+      !is_symbolic_, "Not support symbolic shape now");
+  return this->is_symbolic_ == other.is_symbolic_ &&
+      this->dtype_ == other.dtype_ &&
+      this->device_.type() == other.device_.type() &&
+      this->sizes_ == other.sizes_ && this->strides_ == other.strides_;
```

Comment:
How will comparing the sizes work with symbolic shapes?  I'm a bit confused on how the symbolic shapes strategy works here overall.  I don't think a single `is_symbolic_` bool will be enough.

Since it seems like it isn't fully implemented maybe it would make sense to remove it.

If our goal is to make the first PR simple, we could just rely on the guards (which support symbolic shapes).  We could make the dictionary-based lookup just a static-shapes caching mechanism on top of guards.

cc @ezyang on symbolic shapes

Reply:
The current implementation serves static shapes. I'm working on the symbolic shape support and drafting the PR to illustrate the idea. My basic idea is to store symbolic shapes and sizes as the string like `sizes = [s3, s2, s1, s0], shapes = [s2 * s1 * s0, s1 * s0, s0, 1]` and then build the symbolic variables according to the literals for `sizes_` and `strides_`

Reply:
If it's static, it seems better to be clear about it and store int64_t rather than SymInt

Reply:
Changed it to int64_t.